### ISSUE TITLE
no_proxy not working as expected 

### ISSUE THREAD

**Issue by Carlospi99** [alpinelinux/docker-alpine#292]
I did this

I was using the version 7_85_0 and with the no_proxy flag was taking in account for anything in the list.

no_proxy=local,.example.com,.another.com

But in the version 7_86_0
is only taking in account the last entry of the list, I have change order and it is always the last one that works.
I expected the following
curl/libcurl version

curl 7.86.0 (x86_64-alpine-linux-musl) libcurl/7.86.0 OpenSSL/3.0.7 zlib/1.2.13 brotli/1.0.9 nghttp2/1.51.0
Release-Date: 2022-10-26

No sure if someone had the same

Reported on CURL github too..

https://github.com/curl/curl/issues/9981

**Comment by dlouzan** [comment]
This was indeed a curl issue, and this could be closed as soon as 7.87.0 is stable in alpine https://github.com/curl/curl/issues/9981#issuecomment-1328127020 (currently I see even in edge 7.86.0 https://pkgs.alpinelinux.org/packages?name=curl&branch=edge&repo=&arch=&maintainer=).

**Comment by Carlospi99** [comment]
the new version of curl is already merge and tagged

**Comment by dlouzan** [comment]
@Carlospi99 Yes, I just pulled `latest` and it seems indeed to have the correct version.
